src/riak_kv_wm_queue.erl

```diff
@@ -63,12 +79,12 @@ init(Props) ->
 -spec service_available(#wm_reqdata{}, context()) ->
     {boolean(), #wm_reqdata{}, context()}.
 %% @doc Determine whether or not a connection to Riak
-%%      can be established. Also, extract query params.
+%%      can be established. 
 service_available(RD, Ctx=#ctx{riak=RiakProps}) ->
     ClientID = riak_kv_wm_utils:get_client_id(RD),
     case riak_kv_wm_utils:get_riak_client(RiakProps, ClientID) of
         {ok, C} ->
-            {true, RD, Ctx#ctx{client = C}};
+            {true, RD, Ctx#ctx{client = C, method=wrq:method(RD)}};
```

Comment:
This surprises me... should you really update the context with the method type?

Is that because this is used in malformed_request? If so, I would rather get the method from RD in malformed_request instead of counting on this effect to take place.

Reply:
This may have come about as a copy & paste from https://github.com/basho/riak_kv/blob/develop-3.0/src/riak_kv_wm_object.erl#L222.

I think there is a general problem that I've not been clear about the correct way of working with webmachine, and so this is rather cobbled from other examples from within the riak code.

The same is true with the misuse of malformed request, this is a pattern common elsewhere in riak.

I'm going to look through this agin, and see if it can be better organised.

Reply:
Fixed